stock_cancel/stock.py

```diff
@@ -49,6 +49,10 @@ def action_revert_done(self):
                     _('Picking %s has invoices!') % (picking.name))
             picking.move_lines.write({'state': 'draft'})
             picking.state = 'draft'
+            for move in picking.move_lines:
+                for quant in picking.move_lines.quant_ids:
+                    if move.location_dest_id.id == quant.location_id.id:
```

Comment:
and move.quantity == quant.quantity ? I'm afraid it way not always be the case, because the quant for the transfer of this picking may have add "added" to another quant instead of be created from scratch. In this case, it won't work, no ?
You also need "else raise...."